plugins/flytekit-data-fsspec/flytekitplugins/fsspec/arrow.py

```diff
@@ -47,17 +47,16 @@ def decode(
         self,
         ctx: FlyteContext,
         flyte_value: literals.StructuredDataset,
+        current_task_metadata: StructuredDatasetMetadata,
     ) -> pa.Table:
         uri = flyte_value.uri
         if not ctx.file_access.is_remote(uri):
             Path(uri).parent.mkdir(parents=True, exist_ok=True)
         _, path = split_protocol(uri)
 
         columns = None
-        if flyte_value.metadata.structured_dataset_type.columns:
-            columns = []
-            for c in flyte_value.metadata.structured_dataset_type.columns:
-                columns.append(c.name)
+        if current_task_metadata and current_task_metadata.structured_dataset_type.columns:
```

Comment:
this is no longer optional right?  Can we remove the first if check?

Reply:
yeah, updated it to 
```python
if current_task_metadata.structured_dataset_type and current_task_metadata.structured_dataset_type.columns
```